### PR TITLE
3991 by Inlead: Do not let P2b overtake buy-button.

### PR DESCRIPTION
**Pull request by inlead**
#### Link to issue

https://platform.dandigbib.org/issues/3991

#### Description

When the P2b module is active, it assumes that all ticket-links are given by the P2b module, disallowing the possibility of manual inputting a link to tickets.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: modules/ding_place2book/ding_place2book.module [hunk 1]
```diff
@@ -501,7 +500,7 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
 
       // Catch insert error by checking the event id.
       if (empty($req_result->headers['event-id'])) {
-        $p2b_error = $req_result->headers['status'] . ' - ' . $req_result->status_message;
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: These changes regarding error messages does not seem to be related to the change at hand?

Also: I could not find any documentation of these `error-message` and `error-class` headers anywhere. If we want to use them is there and url we could refer to or the like? A inline comment explaining the situation could also be helpful.

File: modules/ding_place2book/ding_place2book.module [hunk 2]
```diff
@@ -553,7 +552,7 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
       $req_result = drupal_http_request($service_settings['service_url'] . '/delete_event', $options);
 
       if ($req_result->status_message != 'Accepted') {
-        $p2b_error = $req_result->headers['status'];
+        $p2b_error = $req_result->headers['error-class'];
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: See comment above. Also I find it odd that we refer to `error-class` here but `error-message` above.

The context here seems to indicate that we would want an error message.

File: modules/ding_place2book/ding_place2book.module [hunk 3]
```diff
@@ -311,17 +311,16 @@ function ding_place2book_process_node(&$variables) {
...
-      return;
+    $passive = $p2b_settings['passive'];
+
+    if (!empty($p2b_id) && $p2b_id != DING_PLACE2BOOK_NO_ID && !$passive) {
+      $url = !empty($variables['field_ding_event_ticket_link'][0]['url']) ? $variables['field_ding_event_ticket_link'][0]['url'] : '';
+      $variables['book_button'] = theme('place2book_ticketsinfo', [
+        'place2book_id' => $p2b_settings['place2book_id'],
+        'url' => $url,
+        'type' => $p2b_settings['ticket_status'],
+      ]);
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: :+1:

File: modules/ding_place2book/ding_place2book.module (hunk 2 above)
Comment (AnatolyMuntean) [reply modules/ding_place2book/ding_place2book.module]: This was a finding during run-time/debugging. The response headers simply lack the `status` key on those lines and delivered notice errors.
From the implemented logic I assume it tries to get an error string explaining what happened service-wise, so the best value resided in the `error-class` key.

Comment (kasperg) [reply modules/ding_place2book/ding_place2book.module]: The two situations seem very similar. Is there a reason why we would use `error-message` combined with `$req_result->status_message` on L503 and `error-class` (without `$req_result->status_message`) here?

Without any further information I would expect we would use the same key.

Comment (AnatolyMuntean) [reply modules/ding_place2book/ding_place2book.module]: Good question, since I didn't compare whether the service delivers consistent responses for both actions - `/create_event` and `/delete_event`. In other words, whether these - `status_message`, `error-message` and `error-class` - are present and under what circumstance.

File: modules/ding_place2book/ding_place2book.module (hunk 1 above)
Comment (AnatolyMuntean) [reply modules/ding_place2book/ding_place2book.module]: Likewise there was no documentation for the `status`. Either way this gotta be consulted with place2book api. The scope was to fix the notices.
